### ISSUE TITLE
Support git-lfs

### ISSUE THREAD

**Issue by vbjay** [gitextensions/gitextensions#2968]
https://git-lfs.github.com/

**Comment by KindDragon** [comment]
What you mean? It should work as is.
You mean support `git lfs track`?

**Comment by vbjay** [comment]
Right. Show and manage git lfs in the program.

**Comment by wirmachenbunt** [comment]
yeah, we start to use LFS too. a gitextensions integration would be nice (installer & UI)

**Comment by Inspyro** [comment]
Yes. Add an option to the commit window for staging big files to the lfs. (Or do so automatically - by asking for larger files)

**Comment by matkoch** [comment]
:-1:

**Comment by alanedwardes** [comment]
also interested in this :+1:

**Comment by tofutim** [comment]
yes!

**Comment by tofutim** [comment]
Any update on this?

**Comment by KindDragon** [comment]
No one grab it

**Comment by tofutim** [comment]
This is a story about four people named Everybody, Somebody, Anybody and Nobody.  There was an important job to be done and Everybody was sure that Somebody would do it.  Anybody could have done it, but Nobody did it.  Somebody got angry about that, because it was Everybody’s job.  Everybody thought Anybody could do it, but Nobody realized that Everybody wouldn’t do it.  It ended up that Everybody blamed Somebody when Nobody did what Anybody could have.

**Comment by jbialobr** [comment]
The importance of a feature is measured by how many Nobodies are involved in it.
The more Nobodies do the feature the less important it is.

Please, be advised that this software is brought to you free of charge with publicly available source code.

>Any update on this?

What kind of update would you expect?

**Comment by MarcoWilliamsPF** [comment]
I too am interested in this feature..  Has this already been added to GE?

**Comment by KindDragon** [comment]
Nope

**Comment by KindDragon** [comment]
JFI https://developer.atlassian.com/blog/2016/04/git-lfs-12-clone-faster/

**Comment by stefantoller-mohemian** [comment]
(Y)

**Comment by KindDragon** [comment]
Git 2.11 can execute git-lfs fast without using `git lfs` commands https://github.com/blog/2288-git-2-11-has-been-released

**Comment by christianrondeau** [comment]
It would be interesting to hear from someone who knows LFS well what would be expected from this, inside Git Extensions. I'm looking into using Git LFS myself, and I use (and love) Git Extensions, but I'd need more information on what "I want this too" actually includes, and how you see this being implemented. For now, I'll first need to better understand LFS before I do any kind of PR.

**Comment by cdglove** [comment]
@christianrondeau The biggest thing is that, for performance reasons, we typically skip the lfs download step on pull via
```
[filter "lfs"]
	process = git-lfs filter-process --skip
```
in the git config.

This means we need to do a separate pull of LFS to download everything at once. This is a significant speedup, 10x for us, but is based on the number of files you have, their sizes, and your available bandwidth. It would be nice to have this functionality configurable though a UI.

**Comment by vousk** [comment]
I'd also be very pleased to see the ability for GitExtension to automatically call `git lfs checkout/clone/fetch/pull/push/status` instead of `git checkout/clone/fetch/pull/push/status` when it detects that the repo is using Git LFS.
We've just moved our big repos to LFS and standard git commands are really really slow when dealing with LFS files, I barely can work now. Everybody around me is going Atlassian SourceTree but I'd much rather prefer to keep using my beloved GitExtension if possible 🥇 

Can we at least change default command somewhere (to set `git lfs` instead of `git` as default)?

**Comment by yvele** [comment]
Same problem as @vousk , I deployed `git lfs` for my team.. and they mostly use gitextensions.. they don't want to switch to [SourceTree](https://www.sourcetreeapp.com/)

**Comment by KindDragon** [comment]
> I'd also be very pleased to see the ability for GitExtension to automatically call git lfs checkout/clone/fetch/pull/push/status instead of git checkout/clone/fetch/pull/push/status when it detects that the repo is using Git LFS.

They assure that Git 2.11 can execute git-lfs fast without using git lfs commands https://github.com/blog/2288-git-2-11-has-been-released . Why do you need these git lfs commands?

**Comment by vousk** [comment]
Oh sorry, I've just updated GitExtension and it updated git at same time. I thought it would be the latest but I've just checked and git version is now 2.10.1.windows.1, not 2.11. 
I will update git manually to 2.11 and test. I'll keep you informed.

**Comment by cdglove** [comment]
Note: It's not faster by default. You need to add this to your git config

[filter "lfs"]
	process = git-lfs filter-process --skip

But then you need to do an LFS pull manually -- or add a git hook to do it for you on checkout, merge and rebase (which is why it would be nice to configure the UI to do it for you).

**Comment by KindDragon** [comment]
> Note: It's not faster by default. You need to add this to your git config

`git lfs install` should do that

**Comment by cdglove** [comment]
I guess what I said was misleading -- it's faster by default in terms of spawning local processes, but it's not faster in terms of downloading files from LFS; for that you need to tell LFS not to download while processing (that's what the --skip does in "process = git-lfs filter-process --skip"). Otherwise lfs will checkout, and download, each file as it goes which requires establishing a new SSH connection for each file taking roughly 0.3 seconds per file in my tests. Possibly this will be addressed in the future, but for now that it what 's required. Unless someone can tell me some other way to work around that.

**Comment by shcipwa** [comment]
The main workflow issue for me with lfs/extensions is not being able to save files out of commits (view commit, right click save-as) as the smudge filter doesn't seem to run and the file remains as a lfs placeholder.

Is this expected behavior?

**Comment by ltrzesniewski** [comment]
I made a PR which adds an option to clone with LFS: #3834.

It won't close this issue, as other LFS features would need to be supported, but it's an important first step IMO. Cloning using `git lfs clone` instead of `git clone` makes the cloning process much faster for large LFS-enabled repos, and has the added benefit of performing `git lfs install` implicitly.

**Comment by LuqJensen** [comment]
Also interested in this.
Long time fan of this project, so would be sad to have to change <3

**Comment by gerhardol** [comment]
git-lfs is built in to Git, since git-lfs 2.0 the speed is improved and the specific git-lfs-clone command is even deprecated.

A known issue is #6146 (but there is a workaround)

**Comment by GintasS** [comment]
Abandoning this because of age.

**Comment by gerhardol** [comment]
Still relevant
A partial implementation for one specific case in #6189

**Comment by pmiossec** [comment]
> Still relevant
> A partial implementation for one specific case in #6189

Some other work on lfs also in  'support basic lfs operations(lock/unlock and show locks) #8355'
